iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergTimeTravel.java

```diff
@@ -50,10 +49,14 @@ public void testSelectAsOfTimestamp() throws IOException, InterruptedException {
 
     Assert.assertEquals(4, rows.size());
 
-    AssertHelpers.assertThrows("should throw exception", IllegalArgumentException.class,
-        "java.lang.IllegalArgumentException: Cannot find a snapshot older than 1970-01-01", () -> {
-        shell.executeStatement("SELECT * FROM customers FOR SYSTEM_TIME AS OF '1970-01-01 00:00:00'");
-        });
+    try {
```

Comment:
Why is this change?

Reply:
With the change the FetchTask now executed inside the transaction in the Driver, and as a result, if an exception is thrown, it is wrapped by CommandProcessorException. There is a follow-up task: [HIVE-26348](https://issues.apache.org/jira/browse/HIVE-26348) to improve the error message, because now the original reason is not shown to the user